### ISSUE TITLE
Clarification on Microsoft and Azure guideline conflict on naming date time properties

### ISSUE THREAD

**Issue by TimLovellSmith** [Microsoft/api-guidelines#451]
They seem very much in conflict on the topic of ‘createdAt’ vs ‘createdDateTime’, 'lastModifiedAt' vs 'lastModifiedDateTime', 'expiresAt' vs 'expirationDateTime' etc. And the MS guideline is stated as MUST.

Is it deliberate?

[api-guidelines/Guidelines.md at vNext · microsoft/api-guidelines (github.com)](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#17-naming-guidelines)

[api-guidelines/ConsiderationsForServiceDesign.md at vNext · microsoft/api-guidelines · GitHub](https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md#recommended-naming-conventions)

**Comment by JeffreyRichter** [comment]
The OLD guidelines you refer to (1st link you put in your comment) are woefully out of date and have a big notice at the top stating that they should not be used. 

Yes, we deliberately changed the guidance around (not) using DateTime. One reason for the change is because DateTime is very .NET-ish and is not what is used by other languages, and we want Azure REST APIs to be language agnostic to not turn off developers using any language. In addition, the old guidance was not used consistently across Azure services (we researched this) and so it wasn't followed well. "At" was also used pretty heavily and so we went with "At".
